### PR TITLE
Add Transolver model

### PR DESCRIPTION
**Pull request by luohk19**
# Modulus Pull Request

## Description
- Add transolver model from ICML 2024.

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/modulus/blob/main/CHANGELOG.md) is up to date with these changes.
- [x] An [issue](https://github.com/NVIDIA/modulus/issues) is linked to this pull request.

## Dependencies

- einops 0.8.0

### CODE REVIEW

**Comment by mnabian** [comment]
Reference paper: https://github.com/thuml/Transolver/blob/main/Physics_Attention.py

File: examples/cfd/darcy_transolver/train_transolver_darcy.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 - 2024 NVIDIA CORPORATION & AFFILIATES.
```
Comment (mnabian) [inline examples/cfd/darcy_transolver/train_transolver_darcy.py:1]: Is this taken from here? https://github.com/thuml/Transolver/blob/main/Physics_Attention.py
If so, the license header needs to change.

**Comment by mnabian** [comment]
Hi @luohk19 , thanks for your contribution! Have you verified that your implementation reproduces the paper results? If so, could you please post some results here for reference?

File: examples/cfd/darcy_transolver/train_transolver_darcy.py (hunk 1 above)
Comment (luohk19) [reply examples/cfd/darcy_transolver/train_transolver_darcy.py:1]: Yes, I've already changed the license header.

**Comment by luohk19** [comment]
> Hi @luohk19 , thanks for your contribution! Have you verified that your implementation reproduces the paper results? If so, could you please post some results here for reference?

I've discovered some mismatches in problem setting between modulus's darcy and transolver's. In modulus, each batch is a new data simulated by algorithm. But in transolver's setting, training set and test set are fixed. So I fixed the process of generation and successfully reproduces the paper results as follows.

截屏2024-08-15 19 40 56

截屏2024-08-15 19 17 40

**Comment by mnabian** [comment]
Please add unit tests for your datapipe and model, change the license headers and add exceptions, and update the readme to descripe the difference between two training recipes, how to validate the model, and include a plot for the results.

**Comment by luohk19** [comment]
> Please add unit tests for your datapipe and model, change the license headers and add exceptions, and update the readme to descripe the difference between two training recipes, how to validate the model, and include a plot for the results.

I've added unit tests for new datapipe and the transolver, changed the license headers in transolver, and added more details in readme with a plot and a link to the dataset.

**Review by mnabian — APPROVED** [review]
LGTM. Thanks for your contribution!
